### PR TITLE
Nov dev update

### PR DESCRIPTION
**Pull request by korydraughn**

### CODE REVIEW

File: content/posts/irods-development-update-november-2024.md [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+
+
+Hello reader,
+
+We're back from [Supercomputing 2024]({filename}/pages/sc24.html) and it was great! It's always nice to speak to existing and potentially new users.
+
+For the server, we've improved the tracking of ticket information. Exceeding the use-count of a ticket no longer results in an ODBC error. We've also adjusted the ODBC code to return the appropriate iRODS error code when a duplicate key constraint is violated.
+
+Progress on the iRODS 5 server is going well. Two rule engine plugin operations, `setup` and `teardown`, have been added to improve configuration management. These operations speed up agent startup by moving the loading of configuration information to the agent factory. Servers are no longer sensitive to launch order. That means, catalog consumer servers now wait for catalog provider servers to start before accepting client connections, and catalog provider servers now wait for the database to start before accepting client connections. The delay server now locks delay rules before execution to keep other delay servers from duplicating the work. This change helps simplify delay server migration and also sets the foundation for future capabilities.
```
Comment (trel) [inline content/posts/irods-development-update-november-2024.md:14]:
```suggestion
Progress on the iRODS 5 server is going well. Two rule engine plugin operations, `setup` and `teardown`, have been added to improve configuration management. These operations speed up agent startup by moving the loading of configuration information to the agent factory. Servers are no longer sensitive to launch order. That means, catalog service consumers now wait for catalog service providers to start before accepting client connections, and catalog service providers now wait for the database to start before accepting client connections. The delay server now locks delay rules before execution to keep other delay servers from duplicating the work. This change helps simplify delay server migration and also sets the foundation for future capabilities.
```

File: content/posts/irods-development-update-november-2024.md [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+
+Work to remove support for Python 2 from the Python iRODS Client (PRC) is close. The initial work is complete. Python 2 has been EOL for many years, so this is a very important step for the PRC. Support for PAM passwords has improved as well. Users will soon be able to include special characters in their passwords. Aside from that, some of you may have noticed that the **irods** namespace in PyPi was available. The iRODS Consortium has taken ownership of that namespace. Installing the **irods** module via `pip` will now result in the installation of the PRC. To wrap up the PRC update, we have been exploring the possibility of offering a new CLI via the PRC. The goal of this new CLI would be to provide a client that works on all platforms and offer the same power as the iCommands. This effort would take the place of the [C++ CLI](https://youtu.be/wsvvyd0Ot8s?si=CyATYEXuXCWwAXjX) we presented during [UGM 2023](https://irods.org/ugm2023/). We'll have more to share in the future so stay tuned.
```
Comment (trel) [inline content/posts/irods-development-update-november-2024.md:16]:
```suggestion
Work to remove support for Python 2 from the Python iRODS Client (PRC) is close. The initial work is complete. Python 2 has been EOL for many years, so this is a very important step for the PRC. Support for PAM passwords has improved as well. Users will soon be able to include special characters in their passwords. The iRODS Consortium has taken ownership of the **irods** namespace in PyPI. Installing the **irods** module via `pip` will now result in the installation of the PRC. To wrap up the PRC update, we have been exploring the possibility of offering a new CLI via the PRC. The goal of this new CLI would be to provide a client that works on all platforms and offer the same power as the iCommands. This effort would take the place of the [C++ CLI](https://youtu.be/wsvvyd0Ot8s?si=CyATYEXuXCWwAXjX) we presented during [UGM 2023](https://irods.org/ugm2023/). We'll have more to share in the future so stay tuned.
```

File: content/posts/irods-development-update-november-2024.md [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+
+    - draft paper for circulation among group
+    - server-side validation now working
+    - working meeting
+        - Tues, January 21, 2025, 10am ET
+
+- [TRiRODS](https://irods.org/trirods)
+
+    - Wed, December 4, 2024, 4pm ET
+    - speaker, Andrew Su, iRODS Intern
```
Comment (trel) [inline content/posts/irods-development-update-november-2024.md:55]: not sure on this...

File: content/posts/irods-development-update-november-2024.md (hunk 1 above)
Comment (korydraughn) [reply content/posts/irods-development-update-november-2024.md]: Done

File: content/posts/irods-development-update-november-2024.md (hunk 2 above)
Comment (korydraughn) [reply content/posts/irods-development-update-november-2024.md]: Done

File: content/posts/irods-development-update-november-2024.md (hunk 3 above)
Comment (korydraughn) [reply content/posts/irods-development-update-november-2024.md]: Removed for now.

File: content/posts/irods-development-update-november-2024.md [hunk 4]
```diff
@@ -1,5 +1,5 @@
 Title: iRODS Development Update: November 2024
-Date: 2024-11-28 10:00
+Date: 2024-11-27 10:00
```
Comment (trel) [inline content/posts/irods-development-update-november-2024.md:2]: 22:00

**Comment by korydraughn** [comment]
Squashed.

**Pull request merged**